src/client/sse.rs

```diff
@@ -0,0 +1,1099 @@
+            _marker: _,
+        } = self;
+
+        let url = endpoint.map_err(SseError::Network)?;
+        let mut request = reqwest::Request::new(Method::GET, url);
+
+        let headers = request.headers_mut();
+
+        let _ = headers.insert(header::ACCEPT, TEXT_EVENT_STREAM);
+
+        let _ = headers.insert(header::CACHE_CONTROL, NO_STORE);
+
+        if let Some(last_event_id) = last_event_id.header_value() {
+            let _ = headers.insert(LAST_EVENT_ID, last_event_id);
+        }
+
+        // TODO: request's "initiator" type should be set to "other"
+
+        let first_request = match request.try_clone() {
+            None => return Err(SseError::RequestBodyNotCloneable),
+            Some(v) => v,
+        };
+
+        Ok(SseStream {
+            state: SseState::Connecting(client.execute(first_request).boxed()),
+            parser: EventParser::new(
+                request.url().clone(),
+                last_event_id,
+                initial_capacity,
+                max_capacity.unwrap_or(isize::MAX as usize),
+            ),
+            max_retry: max_retry.map(|n| (n, n)),
+            max_loop: max_loop.map(|n| (n, n)),
+            request,
+            client,
+        })
+    }
+}
+
+// SSE CLIENT //////////////////////////////////////////////////////////////////
+
+/// Extension trait for [`Request`]s clients that support subscribing to Server-Sent Events (SSE).
+pub trait SseClient {
+    /// Sends a GET HTTP request to the provided `endpoint`,
+    /// which is expected to serve a stream of Server-Sent Events (SSE).
+    ///
+    /// # Deserialization
+    ///
+    /// `K` and `V` are the types of the expected event's kind and value.
+    ///
+    /// Note that [`Event`]'s kind and value must be valid UTF-8 and must not contain
+    /// newlines (`\r`, `\n` or `\r\n`). If the serial representation of `K` or `V`
+    /// contains newlines, you have to use some sort of compression (like base-64 encoding).
+    ///
+    /// You can use [`String`] to return the original, untyped values.
+    ///
+    /// You can use [`Json<T>`] with any type that implements [`DeserializeOwned`](serde::de::DeserializeOwned).
+    /// It will handle newlines transparently.
+    ///
+    /// # Buffering
+    ///
+    /// `initial_capacity` and `max_capacity` allows to configure the internal
+    /// buffer in which incoming bytes are accumulated before they are decoded.
+    fn sse<K, V>(&self, endpoint: (impl IntoUrl + fmt::Debug)) -> SseStreamBuilder<Self, K, V>
+    where
+        K: FromStr + fmt::Debug + Send + 'static,
+        V: FromStr + fmt::Debug + Send + 'static,
+        Self: Sized;
+
+    /// Same as [`sse`](SseClient::sse) but without deserializing event type and data.
+    fn untyped_sse(&self, endpoint: (impl IntoUrl + fmt::Debug)) -> SseStreamBuilder<Self>
+    where
+        Self: Sized,
+    {
+        self.sse(endpoint)
+    }
+}
+
+impl<T: Request + fmt::Debug + Clone> SseClient for T {
```

Comment:
What do you think about the blanket implementation?

The thing is that now the actual implementation is handled by `StreamBuilder` so there's not much an external implementer could do in there..

Reply:
If we want to keep the trait with methods as shorthand for `SseStreamBuilder::new` with nicer type inference, we do need either one of:
* blanket: `impl<T: Request + fmt::Debug + Clone> SseClient for T`
* `impl SseClient for Client`

IMHO it's fine like this and will benefit all compatible implementors

Reply:
let's keep it like this in that case